CADETProcess/optimization/population.py

```diff
@@ -913,24 +870,29 @@ def update_population(self, population):
             has_twin = False
             to_remove = []
 
-            try:
-                # Do not add if invalid
-                if np.any(np.array(ind_new.cv) > self.cv_tol):
-                    continue
-            except TypeError:
-                pass
+            if not ind_new.is_feasible:
+                continue
 
             for i, ind_pareto in enumerate(self):
                 # Do not add if is dominated
                 if not dominates_one and ind_pareto.dominates(ind_new):
                     is_dominated = True
                     break
+
+                # Remove existing if infeasible
+                elif not ind_pareto.is_feasible:
+                    dominates_one = True
+                    to_remove.append(ind_pareto)
+                    significant.append(True)
+
+                # Remove existing if new dominates
                 elif ind_new.dominates(ind_pareto):
-                    # Remove existing if new dominates
                     dominates_one = True
                     to_remove.append(ind_pareto)
                     if not ind_new.is_similar(ind_pareto, self.similarity_tol):
                         significant.append(True)
+
+                # Ignore similar individuals
                 elif ind_new.is_similar(ind_pareto, self.similarity_tol):
                     has_twin = True
                     break
```

Comment:
Consider defining an else: condition

Reply:
Actually, the `else` is just to continue with the loop and go to the next element.